test/lib/smart-survey-api.test.js

```diff
@@ -2,106 +2,60 @@
-       context.createClient.should.have.been.calledWithExactly('a', 'b');
-    });
-    it('calls this.client.getResponses with the correct params', function() {
-       context.client.getResponses.should.have.been.calledWithExactly(
-        '1', {
-          page: 1, pageSize: 25, includeLabels: false
-        }, context.callback);
-    });
-    it('sets this.client to a new instance of SmartSurveyClient', function() {
-        context = {};
-        // you can mock smartSurveyClientAPI as a constructor if you want to, but not essential
-        smartSurveyClientAPI.createClient.call(context, 'a', 'b');
-        context.client.should.be.instanceof(stubSmartSurvey);
-    });
-    it('calls the SmartSurveyClient constructor', function() {
-        context = {};
-        // reassign the stubSmartSurvey with a stub to listen
-        stubSmartSurvey = sinon.stub();
-
-        // reproxyrequire because the stubSmartSurvey now has a stub
-        smartSurveyClientAPI = proxyquire('../../lib/smart-survey-api', {
-          'smartsurvey-client': stubSmartSurvey
-        });
-
-        smartSurveyClientAPI.createClient.call(context, '1', '2');
-        stubSmartSurvey.should.have.been.called;
-    });
-    it('should call the callback with an error response when we receive an error from the api', function() {
-
-      context = {
-        createClient: sinon.stub(),
-        client: {
-          getResponses: sinon.stub()
-        }
-      };
-      const callback = sinon.stub();
-
-      // Causes the stub to call the argument at the index as a callback function.
-      // stub.callsArg(2); stub calls the 3rd argument as a callback.
-      // callsArgWith like callsArg, but with arguments to pass to the callback.
-      context.client.getResponses.callsArgWith(2, 'error message', 1);
-      smartSurveyClientAPI.getData.call(context, 'a', 'b', undefined, callback);
-      callback.should.have.been.calledWith('error message', 1);
-    });
-    it('should call the callback without an error when we receive success from the api', function() {
-
-      context = {
-        createClient: sinon.stub(),
-        client: {
-          getResponses: sinon.stub()
-        }
-      };
-      const callback = sinon.stub();
-
-      context.client.getResponses.callsArgWith(2, null, 'good');
-      smartSurveyClientAPI.getData.call(context, 'a', 'b', undefined, callback);
-      callback.should.have.been.calledWith(null, 'good');
-    });
+// 2nd argument: An object with the key: module name to proxy & the value the object above
+// So this means smartSurveyStub will now point to this function, i.e. a stub
+const smartSurveyAPI = proxyquire('../../lib/smart-survey-api', { 'smartsurvey-client': smartSurveyStub});
+
+describe('smartSurveyAPI', () => {
+  describe('getData()', () => {
+    it('is a function', () => (typeof smartSurveyAPI.getData).should.equal('function'));
+    // it('is takes 3 arguments', () => (typeof smartSurveyAPI.getData).should.have.lengthOf(3));
```

Comment:
It worked for me. Are you sure that it never worked for you? I'd try to get to the bottom of this...